main/acle.md

```diff
@@ -9382,6 +9415,2344 @@ are named after. All of the functions have external linkage.
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svadd_za32[_f32]_vg1x2(uint32_t slice_base, uint64_t slice_offset,
+                              svfloat32x2_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_f32]
+  //   _za32[_s32]
+  //   _za32[_u32]
+  //   _za64[_f64] (only if __ARM_FEATURE_SME_F64F64 != 0)
+  //   _za64[_s64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svadd_za32[_f32]_vg1x4(uint32_t slice_base, uint64_t slice_offset,
+                              svfloat32x4_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_f32]
+  //   _za32[_u32]
+  //   _za64[_f64] (only if __ARM_FEATURE_SME_F64F64 != 0)
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsub_za32[_f32]_vg1x2(uint32_t slice_base, uint64_t slice_offset,
+                              svfloat32x2_t zm);
+  
+  
+  // Variants are available for:
+  //   _za32[_f32]
+  //   _za32[_u32]
+  //   _za64[_f64] (only if __ARM_FEATURE_SME_F64F64 != 0)
+  //   _za64[_u64] (only if __ARM_FEATURE_SME_I16I64 != 0)
+  __attribute__((arm_streaming, arm_shared_za))
+  void svsub_za32[_f32]_vg1x4(uint32_t slice_base, uint64_t slice_offset,
+                              svfloat32x4_t zm);
+  
+  ```
+
+#### BFCVTN, FCVTN
+
+Multi-vector floating-point convert from single-precision to interleaved half-precision/BFloat16
+
+``` c
+  // Variants are also available for _f16[_f32_x2]
+  __attribute__((arm_streaming))
+  svbfloat16_t svcvtn_bf16[_f32_x2](svfloat32x2_t zn);
+  
+  ```
+
+#### FCVT, BFCVT, FCVTZS, FCVTZU, SCVTF, UCVTF
+
+Multi-vector convert to/from floating-point.
+
+``` c
+  // Variants are also available for _f16[_f32_x2]
+  __attribute__((arm_streaming))
+  svbfloat16_t svcvt_bf16[_f32_x2](svfloat32x2_t zn);
+
+
+  // Variants are also available for _f32[_u32_x2], _s32[_f32_x2] and _u32[_f32_x2]
+  __attribute__((arm_streaming))
+  svfloat32x2_t svcvt_f32[_s32_x2](svint32x2_t zn);
+  
+  
+  // Variants are also available for _f32[_u32_x4], _s32[_f32_x4] and _u32[_f32_x4]
+  __attribute__((arm_streaming))
+  svfloat32x4_t svcvt_f32[_s32_x4](svint32x4_t zn);
+  
+  ```
+
+#### SQCVT, SQCVTU, UQCVT
+
+Multi-vector saturating extract narrow
+
+``` c
+  // Variants are also available for _u16[_s32_x2] and _u16[_u32_x2]
+  __attribute__((arm_streaming))
+  svint16_t svqcvt_s16[_s32_x2](svint32x2_t zn);
+  
+  
+  // Variants are also available for _u8[_s32_x4], _u8[_u32_x4], _s16[_s64_x4],
+  // _u16[_s64_x4] and _u16[_u64_x4]
+  __attribute__((arm_streaming))
+  svint8_t svqcvt_s8[_s32_x4](svint32x4_t zn);
+  
+  ```
+
+#### SQCVTN, SQCVTUN, UQCVTN
+
+Multi-vector saturating extract narrow and interleave
+
+``` c
+  // Variants are also available for _u16[_s32_x2] and _u16[_u32_x2]
```

Comment:
I don't see any instructions for qcvtn with x2 only x4
https://developer.arm.com/documentation/ddi0602/2022-09/SME-Instructions/UQCVTN--Multi-vector-unsigned-saturating-extract-narrow-and-interleave-?lang=en
https://developer.arm.com/documentation/ddi0602/2022-09/SME-Instructions/SQCVTUN--Multi-vector-signed-saturating-unsigned-extract-narrow-and-interleave-?lang=en
https://developer.arm.com/documentation/ddi0602/2022-09/SME-Instructions/SQCVTN--Multi-vector-signed-saturating-extract-narrow-and-interleave-?lang=en

Reply:
Hi @CarolineConcatto, the x2 forms are part of both SVE2p1 and SME2, and are therefore found under the 'SVE instructions' section of that page. They are defined here:

* https://developer.arm.com/documentation/ddi0602/2022-09/SVE-Instructions/SQCVTN--Signed-saturating-extract-narrow-and-interleave-?lang=en
* https://developer.arm.com/documentation/ddi0602/2022-09/SVE-Instructions/UQCVTN--Unsigned-saturating-extract-narrow-and-interleave-
* https://developer.arm.com/documentation/ddi0602/2022-09/SVE-Instructions/SQCVTUN--Signed-saturating-unsigned-extract-narrow-and-interleave-?lang=en